tests/mock_network/churn.rs

```diff
@@ -136,47 +391,57 @@ fn count_nodes_by_section(nodes: &[TestNode]) -> HashMap<Prefix<XorName>, Sectio
     output
 }
 
-/// Adds node per existing prefix using a random proxy. Returns new node indices.
-/// skip_some_prefixes: skip adding to prefixes randomly to allowing sections to merge
-/// when this is executed in the same iteration as `drop_random_nodes`.
-fn add_nodes<R: Rng>(
-    rng: &mut R,
+/// Sub prefix with smaller member counts.
+pub fn sub_perfixes_for_balanced_add(nodes: &[TestNode]) -> BTreeSet<Prefix<XorName>> {
+    let mut counts: BTreeMap<Prefix<XorName>, (usize, usize)> = BTreeMap::new();
+    for node in nodes {
+        let pfx = *node.our_prefix();
+        let name = node.name();
+
+        let (bit_0, bit_1) = counts.entry(pfx).or_default();
+        if name.bit(pfx.bit_count()) {
+            *bit_1 += 1;
+        } else {
+            *bit_0 += 1;
+        }
+    }
+
+    counts
+        .into_iter()
+        .map(|(prefix, (bit_0, bit_1))| {
+            let next_bit_with_fewer_members = bit_0 > bit_1;
+            prefix.pushed(next_bit_with_fewer_members)
```

Comment:
the comment of the function says: an appended bit: `0` if `bit` is `false`, and `1` if `bit` is `true`
which means when bit_0 less than bit_1, a `0` will be appended.
is it expected, or I missed something?

Reply:
yes, we want to add new member in the sub prefix with smallest members to keep things balanced.

so if the count of member in bit_0 is less than the one in bit_1, we add new member to prefix with next bit being 0.

Reply:
ah, right. thx for clarification.